### PR TITLE
feat(brotli): new definition

### PR DESCRIPTION
**Pull request by arturovt**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/brotli/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Type definitions for brotli 1.3
+// Project: https://github.com/foliojs/brotli.js
+// Definitions by: Artur Androsovych <https://github.com/arturovt>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
```
Comment (peterblazejewicz) [inline types/brotli/index.d.ts:5]:
```suggestion
```
the 3.2 is now default, min version, I'm not sure what we could gain here by requesting older version (nothing specific seems to be used)

File: types/brotli/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+import compress = require('./compress');
+import decompress = require('./decompress');
+
+export { compress, decompress };
```
Comment (peterblazejewicz) [inline types/brotli/index.d.ts:10]: If you want, you can follow this TS pattern for exports:
```suggestion
export import compress = require('./compress');
export import decompress = require('./decompress');
```

File: types/brotli/decompress.d.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+/// <reference types="node" />
+
+declare function decompress(buffer: Buffer, outputSize?: number): Uint8Array;
```
Comment (peterblazejewicz) [inline types/brotli/decompress.d.ts:3]:
```suggestion
declare function decompress(buffer: Buffer, outputSize?: number): Uint8Array;
```
it's enough to reference in the `index.d.ts` (I'll add required suggestion snippet).
Simpler to maintain.

File: types/brotli/compress.d.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/// <reference types="node" />
+
+interface CompressOptions {
```
Comment (peterblazejewicz) [inline types/brotli/compress.d.ts:3]:
```suggestion
interface CompressOptions {
```
same, simpler to maintain, I'll add suggestion to `index.d.ts`

File: types/brotli/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+// Type definitions for brotli 1.3
+// Project: https://github.com/foliojs/brotli.js
+// Definitions by: Artur Androsovych <https://github.com/arturovt>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
```
Comment (peterblazejewicz) [inline types/brotli/index.d.ts:4]:
```suggestion
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
/// <reference types="node" />
```
will be used accross DT package, simpler to maintain.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@arturovt thx!

**Review by peterblazejewicz — APPROVED** [review]

**Comment by arturovt** [comment]
Ready to merge

**Pull request merged**